storage/option.go

```diff
@@ -108,3 +111,59 @@ func WithDisabledClientMetrics() option.ClientOption {
 func (w *withDisabledClientMetrics) ApplyStorageOpt(c *storageConfig) {
 	c.disableClientMetrics = w.disabledClientMetrics
 }
+
+// WithDynamicReadReqStallTimeout is an option that may be passed to [NewClient].
+// It enables the client to retry the stalled read request, happens as part of
+// storage.Reader creation. As the name suggest, timeout is adjusted dynamically
+// based on past observed read-req latencies.
+//
+// This is only supported for the read operation and that too for http(XML) client.
+// Grpc read-operation will be supported soon.
+//
+// Here, the input parameter decides the value of dynamic-timeout.
+// targetPercentile is the desired percentile of the observed latencies.
+// increaseRate determines the rate, timeout is adjusted. High means slow increase in
+// timeout and low means rapid increase.
+// initialTimeout decides the initial timeout.
+// minTimeout, maxTimeout is the lower & upper bound of the timeout.
+//
+// TODO (raj-prince): To keep separate dynamicDelay instance for different BucketHandle.
+// Currently, dynamicTimeout is kept at the client and hence shared across all the
+// BucketHandle, which is not not the ideal state. As latency depends on location of VM
+// and Bucket, and read latency of different buckets may lie in different range.
+// Hence hence having a separate dynamicTimeout instance at BucketHandle level will
+// be better
+func WithDynamicReadReqStallTimeout(
```

Comment:
As discussed, we can move this to a storage/experimental subpackage. @frankyn will follow up on this.

Reply:
Done, thanks @frankyn for the  storage/experimental setup support (PR: https://github.com/googleapis/google-cloud-go/pull/10984).